platformio_esp32_envs.ini

```diff
@@ -133,3 +133,35 @@ build_flags               = ${env:test_ESP32_4M316k.build_flags} -DHAS_ETHERNET
 extends                   = env:test_ESP32-wrover-kit_4M316k
 platform                  = ${env:test_ESP32-wrover-kit_4M316k.platform}
 build_flags               = ${env:test_ESP32-wrover-kit_4M316k.build_flags} -DHAS_ETHERNET
+
+
+; A Lolin D32 PRO with 16MB Flash, allowing 4MB sketch size, and 1MB assigned to file storage (at the end of flash)
+[env:max_ESP32_16M1M]
+extends                   = esp32_common
+lib_ignore                = ESP8266WiFi, ESP8266Ping, ESP8266WebServer, ESP8266HTTPUpdateServer, ESP8266mDNS, ESPEasy_ESP8266Ping, ESP32_ping
+board_upload.maximum_size = 4194304
+board_build.partitions    = esp32_partition_app4096k_spiffs1024k.csv
+build_flags               = ${esp32_common.build_flags}  
+                            -DFEATURE_ARDUINO_OTA
+                            -DPLUGIN_BUILD_MAX_ESP32
+board                     = lolin_d32_pro
+
+; A Lolin D32 PRO with 16MB Flash, allowing 4MB sketch size, and 2MB assigned to file storage (at the end of flash)
+[env:max_ESP32_16M2M]
+extends                   = env:max_ESP32_16M1M
+board_build.partitions    = esp32_partition_app4096k_spiffs2048k.csv
+
+; A Lolin D32 PRO with 16MB Flash, allowing 4MB sketch size, and 4MB assigned to file storage (at the end of flash)
+[env:max_ESP32_16M4M]
+extends                   = env:max_ESP32_16M1M
+board_build.partitions    = esp32_partition_app4096k_spiffs4096k.csv
+
+; A Lolin D32 PRO with 16MB Flash, allowing 4MB sketch size, and the rest (nearly 8MB) assigned to file storage
+[env:max_ESP32_16M8124k]
```

Comment:
Maybe rename this one to `env:max_ESP32_16M8M` ?
Don't forget in the Python script for testing resulting file size.

Reply:
> Or did you use the 'k' notation as it isn't a full 8192k size?

Yes, that's the reason.

OT:
I'm now testing with LittleFS (using lorol/LITTLEFS.git as the redirected LittleFS_esp32 throws a weird error during compilation) and I do see it's a tad slow during save, but as SPIFFS is deprecated, I'm thinking of only offering LittleFS variants for the 16MB ESP32's, as there's no backward compatibility required here.

Reply:
> Maybe rename this one to `env:max_ESP32_16M8M` ?

I've now renamed it, and switched to LittleFS for all 16MB environments, will be in the next commit. Much easier to pronounce, and because of the overhead none of the file systems are offering the mentioned size anyway.